crypto/BRCryptoWallet.c

```diff
@@ -40,6 +42,8 @@ cryptoWalletRelease (BRCryptoWallet wallet);
 
 
 struct BRCryptoWalletRecord {
+    pthread_mutex_t lock;
+
```

Comment:
A Wallet-based lock?  I would have thought only a WalletManager-based lock... but a) easy to create locks, b) easy to encapsulate the locks (instead of, for example, needing to pass it down from WalletManager).

Reply:
I've added locks to avoid issues with memory corruption within any of the mutable "classes". Did we want to go beyond that? ETH events are single threaded and ideally BTC will be too so it then becomes a question of do we need to lock for logical integrity.

Reply:
I think it is fine.